app.py

```
@@ -0,0 +1,138 @@
+        region_name=aws_region)
+
+
+# function for sending data to Kinesis at the absolute maximum throughput
+def send_kinesis(kinesis_client, kinesis_stream_name, kinesis_record):
+    kinesisRecords = [kinesis_record]
+    # put the records to kinesis
+    kinesis_client.put_records(
+        Records=kinesisRecords,
+        StreamName=kinesis_stream_name
+    )
+
+
+def parse_kinesis_record(record):
+    try:
+        logger.info("Decoded event with eventId {}".format(record['eventID']))
+        return base64.b64decode(record["kinesis"]["data"]).decode('utf-8')
+    except KeyError as ex:
+        logger.error("Error: Key not found {}".format(ex))
+    except TypeError as ex:
+        logger.error("Type Error: {}".format(ex))
+
+
+def filter_event(record, filter_parameter):
+    try:
+        for element in filter_parameter:
+            values = element.split('.')
+            if len(values) > 1:
+                expr = 'del record'
+                for element_name in values:
+                    expr = expr + '[\'' + element_name + '\']'
+                code = compile(expr, 'delete_record', 'exec')
+                eval(code)
+            else:
+                del record[values[0]]
+        return record
+    except TypeError as ex:
+        logger.error("Error: {}".format(ex))
+
+
+"""
+This method filter the published records in CMSEvent Kinesis Queue.
+
+:param
+-event: The event stream
+"""
+
+
+@app.lambda_function()
+def execute(event, context):
+    # parameters to filter the stream, by default empty
+    to_filter_parameters = os.environ.get('TO_FILTER_PARAMETERS', '')
+    succeeded_record_count = 0
+    failed_record_count = 0
+    skipped_record_count = 0
+    # create a client with kinesis
+    kinesis = create_client('kinesis')
+
+    # if records are not present exit
+    if "Records" not in event:
+        logger.warning("Records not found in the arc cms event stream. Terminating the function")
+        sys.exit(1)
```

You should probably throw an exception here instead. Or is that preferred in lambdas?

---

ya, it can be like error or warn, as after that we are terminating the app.